src/k8s/pkg/snap/util/services.go

```diff
@@ -16,20 +16,20 @@ var (
 		"kube-proxy",
 	}
 	// ControlPlaneServices contains all k8s services that run on a control plane except of k8sd.
-	controlPlaneServices = []string{
+	ControlPlaneServices = []string{
 		"containerd",
-		"kube-apiserver",
 		"kube-controller-manager",
 		"kube-proxy",
 		"kube-scheduler",
 		"kubelet",
+		"kube-apiserver",
```

Comment:
Is kubelet not dependent on the api-server to start first? It needs to register nodes with the api-server?

Reply:
kubelet will wait until the apiserver is up - this is not a problem.

If we would wait until the apiserver is up, we would not avoid the problem as the node-controller could still randomly kick in and kill out snap startup. 
The ~~only~~ preferred way to handle this IMHO is to make sure the controllers do not run before all snap services are settled. 
I'm open for better ideas, though.

Reply:
Actually, waiting for snap services introduces a whole lot of implicit dependencies which we really don't want in the code (see updated description). Let's just retry the hook snap start instead and keep it simple.